### PR TITLE
Add support for promotional collections

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: project_tests.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ set -e
 : "${dependencies:?Need to set dependencies environment variable}"
 if [ "$dependencies" = "lowest" ]; then
     composer update --prefer-lowest --no-interaction
-    proofreader src/
+    proofreader --no-phpcpd src/
```
Comment (giorgiosironi) [inline project_tests.sh:8]: eheh is this trying to tell you something?

File: src/Client/PromotionalCollections.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            ->getPromotionalCollection(
+                ['Accept' => (string) new MediaType(PromotionalCollectionsClient::TYPE_PROMOTIONAL_COLLECTION, self::VERSION_PROMOTIONAL_COLLECTION)],
+                $id
+            )
+            ->then(function (Result $result) {
+                return $this->denormalizer->denormalize($result->toArray(), PromotionalCollection::class);
+            });
+    }
+
+    public function forSubject(string ...$subjectId) : PromotionalCollections
```
Comment (giorgiosironi) [inline src/Client/PromotionalCollections.php]: surprised this isn't a good candidate for an existing or a new trait?

Comment (giorgiosironi) [reply src/Client/PromotionalCollections.php]: This one right? https://github.com/elifesciences/api-sdk-php/pull/204

**Review by giorgiosironi — COMMENTED** [review]
Needs updating from https://github.com/elifesciences/api-sdk-php/pull/204#pullrequestreview-387258689

File: project_tests.sh (hunk 1 above)
Comment (nlisgo) [reply project_tests.sh:8]: From slack:

```
hi chris, let me know when you have time to discuss the api-sdk-php
11:08
The issue is with duplicated code between Collection and RegionalCollection classes
11:08
the normalizer classes are affected also

chris  11:11 AM
...tempted to say just turn off PHPCPD… (edited) 
11:11
Code reviews are more effective, and doesn’t have the strict problem.

nathan:house_with_garden:  11:11 AM
ok, we might hit the same issue with Journal
```

Comment (nlisgo) [reply project_tests.sh:8]: Happy to consider a different approach and avoid the duplication. Would prefer to pair on it to improve chances of acceptance.

Comment (giorgiosironi) [reply project_tests.sh:8]: I think the `ForSubject` trait has addressed most of this duplication. The rest is made up of properties on Collections and PromotionalCollections that are quite similar, but to reduce that we would probably end up extracting traits that don't have a clear name or a domain concept. I'd favor keeping clarity by duplicating those properties, which is the current state of this PR. :+1:

**Review by giorgiosironi — APPROVED** [review]
Functionality and its testing coverage seem complete to me. No further issues for duplication. Open for more reviews.

**Pull request merged**